build_tools/python/e2e_test_framework/definitions/iree_definitions.py

```diff
@@ -225,6 +234,64 @@ def composite_id(self):
         self.input_data.id
     ])
 
+  def materialize_run_flags(self, gpu_id: str = "0"):
+    """Materialize flags with dependent values."""
+    return [
+        flag.replace(E2E_MODEL_RUN_CONFIG_GPU_ID_PLACEHOLDER, gpu_id)
+        for flag in self.composite_flags
+    ]
+
+  @staticmethod
+  def with_flag_generation(module_generation_config: ModuleGenerationConfig,
+                           module_execution_config: ModuleExecutionConfig,
+                           target_device_spec: common_definitions.DeviceSpec,
+                           input_data: common_definitions.ModelInputData):
+    return E2EModelRunConfig(
+        module_generation_config=module_generation_config,
+        module_execution_config=module_execution_config,
+        target_device_spec=target_device_spec,
+        input_data=input_data,
+        composite_flags=generate_run_flags(
+            imported_model=module_generation_config.imported_model,
+            input_data=input_data,
+            module_execution_config=module_execution_config,
+            gpu_id=E2E_MODEL_RUN_CONFIG_GPU_ID_PLACEHOLDER))
+
+
```

Comment:
Moved from `run_module_utils` and merge `build_run_flags_for_model`, `build_run_flags_for_execution_config` into one.